From 23c8c1b2da77adb6653e26b529e16a165b028b33 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:38 -0400
Subject: [tools] perf/lib/traceevent: Add flag to not load event plugins

Message-id: <1410181752-25631-274-git-send-email-jolsa@redhat.com>
Patchwork-id: 91936
O-Subject: [PATCH RHEL7.1 BZ1134356 273/307] tools lib traceevent: Add flag to not load event plugins
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit a7c3196c79051f9e1a498f5be8fe6870bde5e55d
Author: Steven Rostedt (Red Hat) <rostedt@goodmis.org>
Date: Mon Jun 2 23:20:13 2014 -0400

description
===========
Add a flag to pevent that will let the callers be able to set it and
keep the system, and perhaps even normal plugins from being loaded.

This is useful when plugins might hide certain information and seeing
the raw events shows what may be going on.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/event-parse.h b/tools/lib/traceevent/event-parse.h
index feab942..a68ec3d 100644
--- a/tools/lib/traceevent/event-parse.h
+++ b/tools/lib/traceevent/event-parse.h
@@ -354,6 +354,8 @@ enum pevent_func_arg_type {
 
 enum pevent_flag {
  PEVENT_NSEC_OUTPUT  = 1, /* output in NSECS */
+ PEVENT_DISABLE_SYS_PLUGINS = 1 << 1,
+ PEVENT_DISABLE_PLUGINS  = 1 << 2,
 };
 
 #define PEVENT_ERRORS              \
diff --git a/tools/lib/traceevent/event-plugin.c b/tools/lib/traceevent/event-plugin.c
index 0c8bf67..317466b 100644
--- a/tools/lib/traceevent/event-plugin.c
+++ b/tools/lib/traceevent/event-plugin.c
@@ -148,12 +148,17 @@ load_plugins(struct pevent *pevent, const char *suffix,
  char *path;
  char *envdir;
 
+ if (pevent->flags & PEVENT_DISABLE_PLUGINS)
+  return;
+
  /*
   * If a system plugin directory was defined,
   * check that first.
   */
 #ifdef PLUGIN_DIR
- load_plugins_dir(pevent, suffix, PLUGIN_DIR, load_plugin, data);
+ if (!(pevent->flags & PEVENT_DISABLE_SYS_PLUGINS))
+  load_plugins_dir(pevent, suffix, PLUGIN_DIR,
+     load_plugin, data);
 #endif
 
  /*
-- 
1.7.1